### PR TITLE
Update Sarama to version 1.30.0

### PR DESCRIPTION
**Pull request by bai**
Update Sarama and use built-in ParseKafkaVersion, now that it includes all versions.

@dnwe would you mind taking a quick 👀 at this PR?

### CODE REVIEW

**Review by dnwe — COMMENTED** [review]
LGTM

**Comment by jyates** [comment]
This change breaks existing Kafka version parsing. For example, "0.10.2" is no longer an acceptable version string, it has to be 0.10.2.0

File: core/internal/helpers/sarama.go [hunk 1]
```diff
@@ -24,46 +24,9 @@ import (
...
-	"2.5.0":    sarama.V2_5_0_0,
-	"2.6.0":    sarama.V2_6_0_0,
-	"2.7.0":    sarama.V2_7_0_0,
-}
-
 func parseKafkaVersion(kafkaVersion string) sarama.KafkaVersion {
-	version, ok := kafkaVersions[kafkaVersion]
-	if !ok {
+	version, err := sarama.ParseKafkaVersion(kafkaVersion)
+	if err != nil {
```
Comment (jyates) [inline core/internal/helpers/sarama.go:29]: Would it be helpful to include the error in the message so folks know what is wrong?

**Comment by jyates** [comment]
Any thoughts on the comments @bai or @dnwe ?

**Comment by dnwe** [comment]
It's something we could fix in Sarama if you think it would be worthwhile? Did you catch this because you were currently using one of the version numbers that omitted a trailing zero? I believe Kafka moved to semver since 1.0.0 so it's only the 0.x.y.z series that would need this coverage.

**Comment by jyates** [comment]
Yeah, I have a number of clusters that are running older versions. Unfortunate, but such is life when you don't control everything. Been running burrow for years now with the old version (missing the trailing .0), so this was just a bit surprising.

I'm fine with the change, if you would prefer _not_ to fix this (sarama approach looks reasonable). Just wanted to raise it as "breaking" change that is probably worth mentioning to users.

Or a hybrid approach could be Ok too - try the sarama version lookup, if that fails, fall back to the old version mapping for pre-1.0.0 versions.

**Pull request merged**